### DISCUSSION TITLE
Public beta feedback

### DISCUSSION THREAD

**Discussion by spwoodcock** [hotosm/field-tm#1364] [General]
### Lessons From Monrovia (Liberia) Slum Assessment

- Splitting algorithm needs to work without polylines provided, e.g. from a custom extract.
    - Especially important for areas with no roads or rivers.
    - Just divide the buildings by the number specified as best as possible.
    - The custom extract cannot be passed through as the splitting extract if we do not fix the problem above
    - We also need configuration to allow the user to have data extract for mapping, and data extract for splitting
- As a consequence of the above, we probably need two separate project creation flows:
    - Custom flow: custom form, custom extract, everything custom.
    - Basic flow: use OSM extract, provided XForm, etc.
- We can manually append required properties to the geojson if they don't exist in a custom data extract (DONE & in production)
    - Managing flatgeobuf data extracts is more difficult
- Bug: XForm title needs task id in it to display correctly in ODK Collect (the user must know the task ID they are mapping).
- Not all users are OSM'ers. We need to have another possibility for login. Email would be ok.
    - A login token could be provided from the backend, essentially providing authorization, but no authentication (as only an email address is required). This is ok for the mapper flow only (i.e. basic usage of FMTM).

@manjitapandey @susmina94 @ivangayton feel free to add any extra feedback here.

**Comment by spwoodcock** [comment]
### Feedback from discussions at NAXA

- We need a good way to digitize buildings that do not exist.
- Proposed flow:
  - TM is used to generate a training dataset from imagery (i.e. 10% of buildings mapped).
  - fAIR integration for FMTM to upload the trained data geojson, then generate a model.
  - This model ID is used in the fAIr predictor API to generate geojson building predictions for the entire area.
    - Predictor API requires model ID and AOI.
- We should add auth to osm-rawdata and a path prefix in format /fmtm/(dev/stage/prod)/project_id
  - This will persist data extracts for more than 90 days.
  - It also allows for status of downloads to be collected from raw-data-api access logs.

**Comment by spwoodcock** [comment]
### Tasking Manager for training data
- We need a tasking manager instance that is **sandboxed** from OSM
- This allows us to generate training data for an area collaboratively
- If using OSM, the first step is to delete all the existing geoms, which isn't ideal (iD editor does not have a way to do this easily + we may lose existing geometry tag data.
- The final step of digitisation would then be conflation with the existing OSM data, which we have workflows for.

### Division of tasks for fAIr

- Representative areas from the base imagery should be chosen for digitisation for training data.
- Either only upload a small subset of the AOI to TM, or the task division needs to be coordinated amongst the team (i.e. map this task, don't map this task).

### Loading TMS default for project

- Currently each user must have the custom TMS URL for base imagery (drone data), and specify this in a Custom Layer in the layer switcher.
- This isn't very intuitive - ideally we can have the layer preloaded for a project, so users don't have to configure themselves.
- The configuration is likely done within iD editor?

### Validating fAIr output
- We ideally need to pass the split buildings for fAIr back to a TM instance for splitting of lumped geoms, etc (not possible yet).
- Feedback can be provided collaboratively by a team.
- We need an easy way to split - in QGIS this would be via split tool. But can iD editor do this?
- We should probably split geoms by roof outline for consistency (the default of OSM currently), instead of building footprint (the preferred approach into the future).

### Input to FMTM
- The output from this process goes into FMTM.
- Uploaded as a custom data extract.

### Refinement to FMTM flow
- Improved mapper workflow with orientation and geolocation ability.
- Loading basemaps into offline web browser storage should be more prominent / intuitive.
- Adding routing is a nice to have, but not essential for now.
- Improve workflow via Entities: buildings marked as in progress, or complete.
- Extraction of data from FMTM should be improved significantly.

### XLSForms in FMTM
- We need solid XLSForm base for users to extend from - they should have most things filled by default, where the user can then add their specific questions below.
- Easily downloadable via FMTM.
- Entirely custom XLSForms should be possible, but will be more difficult to manage / edge case

**Comment by spwoodcock** [comment]
## Cligeos Nepal Feedback

- One major thing we noticed is whenever we have tasks and people go on field. We were approx 30 people trying at the same time now since QR is in first step and after QR is downloaded we need to lock the task while doing so somebody else locks that task already and now forms are duplicated. QR should be generated only after task is locked.
  - **Action**: only show QRCode after 'Start Mapping' button is clicked.
- Sign in sign out is not working properly , apps says signed in by default, lot of people got confused.
  - Related issue https://github.com/hotosm/fmtm/issues/1289
  - @spwoodcock will work on this today.
- Concerns raised with arcgis tiles in map, why not render osm tiles by default.
  - @varun2948 should we load OSM tiles by default?
- Mbtiles generation, supports custom TMS URL, it would be good to have option to enter custom tms url in the project creation phase directly.
  - Related PR for backend part https://github.com/hotosm/fmtm/pull/1446
  - @NSUWAL123 working on the frontend part.
- It would be good if we can load the global contour data, since its a field mapping activity it would be nice to consider the elevation data or at least we can see it int he basemap while assigning task to people, so people who can walk we can give them based on contour difference.
  - **Action**: add a global DEM layer from cloud optimised geotiff.
- Number of contributors is different than home screen and different in tasks page (BUG)
  - **Action**: Issue created https://github.com/hotosm/fmtm/issues/1451
- User-friendly error message on XLS form validation, Easy interactive way of XLS form creation method. It is very difficult to figure out what went wrong in my XLS form. It simply says xls is invalid it doesn’t say why. And it would be lot more useful if we can interactively delete / add fields in form like KOBO within FMTM. Let's say i want to select building preset remove some unnecessary questions and add a couple of on my own, it would be lot more easier within UI
  - **Note**: This is rather tricky. Enketo does this sort of thing. We could definitely do a simple version, but a rich editor is out of scope of FMTM I would say.
- Access token should be marked as required here https://api.fmtm.hotosm.org/docs#/submission/download_submission_submission_download_get
  - **Note** the access-token param is just an artifact in the docs, but isn't param that should be provided manually. It's extracted from the users login cookie, so done by default. Having the param present means that login is required:
  - ![image](https://github.com/hotosm/fmtm/assets/78538841/588855e4-4e0d-4559-a732-d6fb8ec35f6a)
  - So this is all good 👍 
- We should at least have working UI button to download form submission as geojson before performing e2e exercise, otherwise it would be hard to showcase the result. We did had workarounds with joins and custom python script. I gather that this is work in progress. Very important feature.
  - **Action**: added a working endpoint to do this https://api.dev.fmtm.hotosm.org/docs#/helper/convert_odk_submission_json_to_geojson_wrapper_helper_convert_odk_submission_json_to_geojson_post
  - The logic should be moved into the submissions router and linked to a button on the frontend
  - Created issue https://github.com/hotosm/fmtm/issues/1452
- Basemap problem, couldn’t figure which building is which, no compass sensor ! ( This could be ODK problem but it is major disadvantage specially if buildings are close ).
  - **Note** this is addressed by the mobile mapper experience in FMTM / Entities workflow in https://github.com/hotosm/fmtm/pull/1449
  - The user should do all navigation in FMTM, then load the form with the feature pre-selected in ODK.
- Name of people who locked the tasks, it matters when we are in the field but might not make sense in tasking manager. So that we can group together and head down.
  - **Note**: please clarify?
- No tutorials about how to load mbtiles in ODK. We couldn’t do it, if there was still offline osm vector tiles that would have done the job too.
  - **Note** this will be improved once https://github.com/getodk/collect/pull/5917 is restarted. 
  - We need documentation for this though! @manjitapandey 
- Lot of people suggested that they locked their task, but they couldn't find it easily in FMTM it would be nice to have task locked by you or different color for my tasks. It is visible in ODK itself which is nice.
  - **Action** created issue https://github.com/hotosm/fmtm/issues/1453
- During mapping on ODK, how to know which building is done and which isn’t? This is also major bottleneck for us if there was an option lets say if mapped building turns to green or something it would be lot easier.
  - **Note**: work in progress in https://github.com/hotosm/fmtm/pull/1449
  - Will be out in next release.
- Pushing field data to OSM ( we haven’t performed this one ) 
  - **Note**: longer term goal!
- Difficult to edit the submission, some users mistakenly pushed some forms, how to edit them? Or delete some.
  - **Note**: I think it's possible to edit forms that have not been synced to ODK yet.
  - We should investigate the different options to help with this.

**Comment by kshitijrajsharma** [comment]
>Name of people who locked the tasks, it matters when we are in the field but might not make sense in tasking manager. So that we can group together and head down. 

Meaning if there is a way to know who locked which tasks or tasks can be labeled on the basis of names of mappers it would be easy to form group and distribute to people who are going in same direction , Here : 
May not be ideal , I am not sure this is one of the suggestion I got from group 
<img width="548" alt="image" src="https://github.com/hotosm/fmtm/assets/36752999/4e000e1c-51bd-4798-86d4-e1b17e457116">

**Reply by varun2948** [reply]
we initially had task id on the task centroid but if we get ahead on entities stuff with the metadata we could surely have it on the centroid of the task of who the task has been assigned to.

**Comment by spwoodcock** [comment]
Notes on `field admin` role:
- Field admin needs to approve site done at end possibly?
- Or possible dashboard for them to see missing buildings?
- We have went some way to address this validation by adding a confirmation dialog to the user if they 'mark complete', but not all features are mapped: https://github.com/hotosm/fmtm/pull/1493
